docs/content/preview/explore/fault-tolerance/transaction-availability.md

```diff
@@ -0,0 +1,234 @@
+can kill the correct node to see what is happening. For this one needs to get the hash code of the primary key using the `yb_hash_code(id)` function and co-relate with the `yb-admin` output to figure out where that 
+key is located. For the examples we will focus on the row with `k1=1`.
+
+```sql
+SELECT id, upper(to_hex(yb_hash_code(id))) AS hash FROM generate_series(1,5) AS id;
+```
+```output.sql
+ id | hash
+----+------
+  1 | 1210
+  2 | C0C4
+  3 | FCA0
+  4 | 9EAF
+  5 | A73
+(5 rows)
+```    
+List the nodes via 
+```sh
+yb-admin list_tablets ysql.yugabyte txndemo
+```
+
+```output.sh
+Tablet-UUID                        Key Range         Leader-IP       Leader-UUID
+7e2dfb66a4654aa5b2fb133b446aaabc   [0x0000, 0x5554]  127.0.0.2:9100  4739b43f76184e1cab003b88686df290
+a9b4675fdaaa4d4b949adc5e53d183bf   [0x5555, 0xAAA9]  127.0.0.3:9100  7402fbc9c6384d80bb7bcd09b89dbca9
+a8c50129f63642459a02ed4ee492a1f3   [0x0000, 0x5554]  127.0.0.1:9100  6789e52b1c334844a66078fe9fdf95fa
+```
+ 
+From the hash ranges listed on the tablet-servers page at <http://localhost:7000/tablet-servers> and [yb-admin](../../../admin/yb-admin/) output, we can see that the row with `k1=1` whose hash code is `1210` resides on node `127.0.0.2` as that node has the tablet which contains the key range `[0x0000, 0x5554]`. 
+
+
+## Failure Scenario 1
+In this example, you are going to see how a transaction successfully completes when the node that has just received a [provisional write](../../../architecture/transactions/distributed-txns/#provisional-records) fails.
+
+Connect to `127.0.0.3` and start a simple transaction to update the value of row `k1=1` to `20`.
+```output.sql
+ysqlsh -h 127.0.0.3  -U yugabyte -d yugabyte
+yugabyte@yugabyte=# BEGIN;
+BEGIN
+Time: 2.047 ms
+yugabyte@yugabyte=# UPDATE txndemo set k2=20 where k1=1;
+UPDATE 1
+Time: 51.513 ms
+```    
+Stop the node:`127.0.0.2`. _Note_: We are stopping `127.0.0.2` as that is the node that contains `k1=1` which we figured from the steps in the [Identifying the key location](#identifying-the-key-location) section.
+```sh
+yugabyted stop --base_dir=/tmp/ybd2
+```
+
+List the nodes via 
+```sh
+yb-admin list_tablets ysql.yugabyte txndemo
+```
+
+```output.sh
+Tablet-UUID                        Key Range         Leader-IP       Leader-UUID
+7e2dfb66a4654aa5b2fb133b446aaabc   [0x0000, 0x5554]  127.0.0.3:9100  7402fbc9c6384d80bb7bcd09b89dbca9
+a9b4675fdaaa4d4b949adc5e53d183bf   [0x5555, 0xAAA9]  127.0.0.3:9100  7402fbc9c6384d80bb7bcd09b89dbca9
+a8c50129f63642459a02ed4ee492a1f3   [0x0000, 0x5554]  127.0.0.1:9100  6789e52b1c334844a66078fe9fdf95fa
+```
+Notice that `127.0.0.2` is gone from the list and `127.0.0.3` is now the leader for the tablet that was in `127.0.0.2`(`7e2dfb66a..`)
+
+Commit the transaction.
+```output.sql
+yugabyte@yugabyte=# COMMIT;
+COMMIT
+Time: 6.243 ms
+```    
+We can see that even though the node `127.0.0.2` failed after receiving the provisional write, the transaction has succeeded.
```

Comment:
might be good to add a step where we read the table after commit and confirm the data is there

Reply:
Good point. I had felt that showing the `COMMIT` succeeding was enough. But yes, I'll add the confirmation of data.